frontend/src/metabase/lib/expressions/compile.js

```diff
@@ -76,21 +76,22 @@ class ExpressionMBQLCompilerVisitor extends ExpressionCstVisitor {
     }
     return ["metric", metric.id];
   }
-  segmentExpression(ctx) {
-    const segmentName = this.visit(ctx.segmentName);
-    const segment = parseSegment(segmentName, this._options);
-    if (!segment) {
-      throw new Error(`Unknown Segment: ${segmentName}`);
-    }
-    return ["segment", segment.id];
-  }
   dimensionExpression(ctx) {
```

Comment:
Postponing the dimension/segment distinction until after initial parsing makes sense to me. Though, if we're doing that should we call that unresolved item something else? If there's no better name, maybe just "or-ing" the options in the name `dimensionOrSegment`?

Reply:
My plan is eventually to get rid of this completely (no dimension/segment/metric during parsing) after some more PRs since as the syntax level, they are all just identifiers. Only after resolving them correctly after the parsing then we can know correctly the semantics of each identifer.